hack/.ci/run-e2e-gke.sh

```diff
@@ -172,26 +174,96 @@ ingress_controller_address="$( kubectl -n haproxy-ingress get svc haproxy-ingres
 
 kubectl create -n e2e pdb my-pdb --selector='app=e2e' --min-available=1 --dry-run=client -o yaml | kubectl_create -f -
 
-kubectl -n e2e run --restart=Never --image="${SO_IMAGE}" --labels='app=e2e' --command=true e2e -- bash -euExo pipefail -O inherit_errexit -c "function wait-for-artifacts { touch /tmp/done && until [[ -f '/tmp/exit' ]]; do sleep 1; done } && trap wait-for-artifacts EXIT && mkdir /tmp/artifacts && scylla-operator-tests run '${SO_SUITE}' --loglevel=2 --color=false --artifacts-dir=/tmp/artifacts --feature-gates='${SCYLLA_OPERATOR_FEATURE_GATES}' --ingress-controller-address='${ingress_controller_address}' --ingress-controller-ingress-class-name='${ingress_class_name}' --ingress-controller-custom-annotations='${ingress_custom_annotations}' --scyllacluster-node-service-type='${SO_SCYLLACLUSTER_NODE_SERVICE_TYPE}' --scyllacluster-nodes-broadcast-address-type='${SO_SCYLLACLUSTER_NODES_BROADCAST_ADDRESS_TYPE}' --scyllacluster-clients-broadcast-address-type='${SO_SCYLLACLUSTER_CLIENTS_BROADCAST_ADDRESS_TYPE}'"
+gcs_service_account_credentials_container_arg=
+if [[ -n ${SO_GCS_SERVICE_ACCOUNT_CREDENTIALS_PATH+x} ]]; then
+  gcs_service_account_credentials_container_arg=$(cat << EOF
+    - --gcs-service-account-key-path=/var/run/secrets/gcs-service-account-credentials/gcs-service-account.json
+EOF
+)
+
+  kubectl create -n e2e secret generic gcs-service-account-credentials --from-file="${SO_GCS_SERVICE_ACCOUNT_CREDENTIALS_PATH}" --dry-run=client -o yaml | kubectl_create -f -
+else
+  kubectl_create -n e2e secret generic gcs-service-account-credentials
+fi
+
+e2e_pod=$(cat << EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: e2e
+  name: e2e
+spec:
+  restartPolicy: Never
+  containers:
+  - name: wait-for-artifacts
+    command:
+    - bash
+    - -euExo
+    - pipefail
+    - -O
+    - inherit_errexit
+    - -c
+    - function wait-for-artifacts { touch /tmp/done && until [[ -f '/tmp/exit' ]]; do sleep 1; done } && trap wait-for-artifacts EXIT
+    image: ${SO_IMAGE}
+    imagePullPolicy: Always
+    volumeMounts:
+    - name: artifacts
+      mountPath: /tmp/artifacts
+  - name: e2e
+    command:
+    - scylla-operator-tests
+    args:
+    - run
+    - ${SO_SUITE}
+    - --loglevel=2
+    - --color=false
+    - --artifacts-dir=/tmp/artifacts
+    - --feature-gates=${SCYLLA_OPERATOR_FEATURE_GATES}
+    - --ingress-controller-address=${ingress_controller_address}
+    - --ingress-controller-ingress-class-name=${ingress_class_name}
+    - --ingress-controller-custom-annotations=${ingress_custom_annotations}
+    - --scyllacluster-node-service-type=${SO_SCYLLACLUSTER_NODE_SERVICE_TYPE}
+    - --scyllacluster-nodes-broadcast-address-type=${SO_SCYLLACLUSTER_NODES_BROADCAST_ADDRESS_TYPE}
+    - --scyllacluster-clients-broadcast-address-type=${SO_SCYLLACLUSTER_CLIENTS_BROADCAST_ADDRESS_TYPE}
+    - --object-storage-bucket=${SO_BUCKET_NAME}
+${gcs_service_account_credentials_container_arg}
```

Comment:
```suggestion
    - --gcs-service-account-key-path=${gcs_sa_in_container_path}
```
if the variable is `""` then this won't effectively be set but you can keep a nicer structure where you don't have to worry about indentation

Reply:
done